File: csharp/O/After/Strategy/BreakerStrategy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace After.Strategy;

public class BreakerStrategy : IProtectionStrategy
{
    public double CalcBreakingCapacity()
    {
        return 1;
    }

    public double CalcIn()
    {
        return 1;
    }

    public bool CheckProtection()
    {
        return true;
    }
}

File: js-ts/O/after/electrical-protection/Strategy/BreakerStrategy.ts
import { IProtectionStrategy } from "./IProtectionStrategy.ts";

export class BreakerStrategy implements IProtectionStrategy {
    public calcIn(In: number): number {
        return In + 1;
    }

    public calcBreakingCapacity(): number {
        return 1;
    }

    public checkProtection(): boolean {
        return true;
    }
}
